index.js

```diff
@@ -37,12 +37,18 @@ function compressPlugin (fastify, opts, next) {
   const supportedEncodings = ['gzip', 'deflate', 'identity']
   if (opts.brotli) {
     compressStream.br = opts.brotli.compressStream
-    supportedEncodings.push('br')
+    supportedEncodings.unshift('br')
   } else if (zlib.createBrotliCompress) {
     compressStream.br = zlib.createBrotliCompress
-    supportedEncodings.push('br')
+    supportedEncodings.unshift('br')
   }
 
+  const encodings = Array.isArray(opts.encodings)
+    ? supportedEncodings
+      .filter(encoding => opts.encodings.includes(encoding))
+      .sort((a, b) => opts.encodings.indexOf(a) - supportedEncodings.indexOf(b))
+    : supportedEncodings
```

Comment:
I think we should throw if this array is empty

Reply:
I’m having trouble figuring out how I should test this. This doesn’t pass, for example:

```js
test('Should throw if `encodings` array is empty', t => {
  const fastify = Fastify()
  t.throws(
    fastify.register(compressPlugin, { encodings: [] }),
    {}
  )
})
```

Reply:
if encoding is empty, calll the callback with an error.

Reply:
`t.throws` expects a function as the first parameter: https://node-tap.org/docs/api/asserts/#tthrowsfn-expectederror-message-extra

```js
test('Should throw if `encodings` array is empty', t => {
  const fastify = Fastify()
  t.throws(() => {
    fastify.register(compressPlugin, { encodings: [] })
  })
})
```